### PR TITLE
Centralize the generate docindex scripts.

### PR DESCRIPTION
**Pull request by sima-zhu**

### CODE REVIEW

File: eng/common/scripts/Generate-DocIndex.ps1 [hunk 1]
```diff
@@ -0,0 +1,150 @@
+# Generates an index page for cataloging different versions of the Docs
+[CmdletBinding()]
+Param (
+    $DocFx,
+    $RepoRoot,
+    $DocGenDir
+)
+. (Join-Path $PSScriptRoot common.ps1)
+$GetGithubIoDocIndexFn = "Get-${Language}-GithubIoDocIndex"
```
Comment (weshaggard) [inline eng/common/scripts/Generate-DocIndex.ps1:9]: This is exactly the change I had in mind when we talked but my expectation is that we wouldn't put the changes under eng/common yet. Instead we should just put the script in each of the language repos and get those changes checked in there and once we feel good about them working there then we can refactor and put this script in eng/common.

Comment (sima-zhu) [reply eng/common/scripts/Generate-DocIndex.ps1:9]: You mean we put `$GetGithubIoDocIndexFn = "Get-${Language}-GithubIoDocIndex"` in Language-Setting.ps1, right?
Done with the changes. Might look a weird for people when they looked into language-Setting script.

Comment (weshaggard) [reply eng/common/scripts/Generate-DocIndex.ps1]: No I was meaning to put Generate-DocIndex.ps1 in the language repo not eng/common yet. I was wanting this to be duplicated in every language repo until we have been able to see that it works OK for all those repos and then refactor and move it under eng/common. Once you move this under eng/common then you can split it out into the correct common scripts.

Comment (sima-zhu) [reply eng/common/scripts/Generate-DocIndex.ps1:9]: Does it overwrite every time we merge sync PR?

**Comment by weshaggard** [comment]
Closing this PR for now to free up resources in our DevOps. Once we are ready to move this into eng/common we can re-open a version of this PR.

**Pull request closed without merging**